internal/pkg/api/plugin.go

```diff
@@ -0,0 +1,94 @@
+//                                            -------------------------------------------
+//                             				   ||
+//                                             ||        --------------------------------
+//                                             ||     ->|         Sharing Client        |
+//                                             ||    |   --------------------------------
+//                                             ||    |
+//                                             \/    |            SegmentSender
+//                                            ---    |  ---------------------------------
+//                                           |   |   | |  -----------       -----------  |
+//                                           | D |   |-| |BatchBuffer| ==> | Forwarder | |
+//                                           | i |   | |  -----------       -----------  |
+//                                           | s |   |  ---------------------------------
+//                                           | p |   |
+//                                           | a |=> |              .......                 ===> Kafka/OAP
+//                                           | t |   |
+//                                           | c |   |             MeterSender
+//                                           | h |   | -----------------------------------
+//                                           | e |   -|  -------------       -----------  |
+//                                           | r |    | | BatchBuffer | ==> | Forwarder | |
+//                                           |   |    |  -------------       -----------  |
+//                                            ---      -----------------------------------
+//
+//
+// 1. The Collector plugin would fetch or receive the input data.
+// 2. The Parser plugin would parse the input data to SerializationEvent that is supported
+//    to be stored in Queue.
+// 3. The Queue plugin stores the SerializationEvent. However, the serialization depends on
+//    the Queue implements. For example, the serialization is unnecessary when using a Memory
+//    Queue.
+// 4. The Filter plugin would pull an event from the Queue and process the event to create
+//    a new event. Next, the event is passed to the next filter to do the same things until
+//    the whole filters are performed. The events labeled with RemoteEvent type would be
+//    stored in the OutputEventContext. When the processing finished,
+//    the OutputEventContext. After processing, the events in OutputEventContext would be
+//    partitioned by the event type and sent to the different BatchBuffers, such as Segment
+//    BatchBuffer, Jvm BatchBuffer, and Meter BatchBuffer.
+// 5. When each BatchBuffer reaches its maximum capacity, the OutputEventContexts would be
+//    converted to BatchEvents and sent to Forwarder.
+// 6. The Follower would send BatchEvents and ack Queue when successful process this batch
```

Comment:
How to define the output events which could have suitable senders?

Reply:
Or could I say, a follower match one type of event?

Reply:
![image](https://user-images.githubusercontent.com/31562192/100545597-8b182d00-3297-11eb-8bb6-242b0c90bb5f.png) this mapping would be contained in Dispatcher

Reply:
add comments in plugin.go

Reply:
add eventType on forwarder